src/ToolBox/SOS/lldbplugin/inc/lldbservices.h

```diff
@@ -237,6 +237,9 @@ MIDL_INTERFACE("2E6C569A-9E14-4DA4-9DFC-CDB73A532566")
 ILLDBServices : public IUnknown
 {
 public:
+
+    virtual ~ILLDBServices() = default;
```

Comment:
Seems kind of odd to have a destructor in a COM interface. Well, this is probably not real COM but still, it inherits `IUnknown`. Is there actually code that uses delete on a ILLDBService pointer?

Reply:
Here's the error from clang:

```
/home/omajid/devel/dotnet/coreclr/src/ToolBox/SOS/lldbplugin/services.cpp:68:9: error: delete called on non-final    
      'LLDBServices' that has virtual functions but non-virtual destructor [-Werror,-Wdelete-non-virtual-dtor]       
        delete this;
        ^
1 error generated.
```

Reply:
I think that the proper solution here is to make `LLDBServices`'s destructor virtual or make the class final. Nobody's going to use `delete` on a COM interface (there's no `ILLDBService* p = … ; delete p;` anywhere in the code)

Reply:
I totally mixed up `ILLDBService` and `LLDBService` :( Fixed now.